Chore/test coverage/participation resource

maxr96 opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
- [x] Server: I added multiple integration tests (Spring) related to the features

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
This PR increases the coverage of Participation resource from 32% to 90% . All REST endpoints are covered now.
![Screenshot_2020-02-25 de tum in www1 artemis web rest](https://user-images.githubusercontent.com/17262376/75233361-07d22400-57b9-11ea-8be3-9fb173511e95.png)

maxr96 commented on src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseParticipationService.java:
```diff
@@ -111,13 +111,18 @@ public TemplateProgrammingExerciseParticipation findTemplateParticipationByProgr
         }
+        var participation = optionalParticipation.get();
+        // If the exercise associated with participation is not programming exercise return null
+        if (participation.getProgrammingExercise() == null) {
```
I check before to ensure that we don't have uncaught NullPointerException

krusche replied:
please revert, your test case is wrong if you experience a null pointer exception. Also returning null would lead to other problems here

maxr96 replied:
Reverted

maxr96 commented on src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java:
```diff
@@ -196,15 +196,10 @@ public ParticipationResource(ParticipationService participationService, Programm
 
         ProgrammingExerciseStudentParticipation participation = programmingExerciseParticipationService.findStudentParticipationByExerciseIdAndStudentId(exerciseId,
                 principal.getName());
-        if (participation == null) {
```
This part was never reachable before since we checked in the method above is optional is empty and throw an exception.

krusche replied:
please revert and use the new version of develop

maxr96 replied:
Done

maxr96 commented on src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java:
```diff
@@ -215,21 +210,20 @@ public ParticipationResource(ParticipationService participationService, Programm
+            participation.getExercise().filterSensitiveInformation();
+            return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, participation.getStudent().getName())).body(participation);
+        }
+        else {
```
Only in the case if participation is null, which means that exercise attached to the participation is not of Programming Exercise type we reach this part

krusche replied:
revert

maxr96 replied:
Done

maxr96 commented on src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java:
```diff
@@ -561,12 +550,12 @@ else if (quizExercise.isSubmissionAllowed()) {
-    public ResponseEntity<?> getParticipationStatus(@PathVariable Long id) {
-        StudentParticipation participation = participationService.findOneStudentParticipation(id);
+    @GetMapping("/participations/{participationId}/status")
+    public ResponseEntity<?> getParticipationStatus(@PathVariable Long participationId) {
```
The endpoint was not reachable as it could not map the value from the path, so I changed it

krusche replied:
please delete the whole end point

maxr96 replied:
I merged this changed from the develop

maxr96 commented on src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java:
```diff
@@ -267,13 +257,12 @@ private void addLatestResultToParticipation(Participation participation) {
         if (participation.getId() == null) {
             return createParticipation(participation);
         }
+        if (participation.getPresentationScore() == null || participation.getPresentationScore() < 0) {
```
This should occur first to catch nulls

krusche replied:
ok

krusche commented on src/test/java/de/tum/in/www1/artemis/ParticipationIntegrationTest.java:
```diff
@@ -67,11 +87,21 @@ public void initTestCase() {
     @AfterEach
     public void tearDown() {
         database.resetDatabase();
+        Feature.PROGRAMMING_EXERCISES.enable();
```
good! this change is also needed in the other PR

maxr96 replied:
Added this to the other PR as well

maxr96 commented:
> see inline comments and please merge develop

I resolved the inline comments, removed not used tests and added a couple of new test cases. Also merged developed.
